frontend/src/components/CreateEngagementModal/index.tsx

```diff
@@ -0,0 +1,132 @@
+import 'toastr/build/toastr.min.css';
+
+import Button from '@material-ui/core/Button';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import Clear from '@material-ui/icons/Clear';
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Flatpickr from 'react-flatpickr';
+
+import { useAuth0 } from '../../auth0Spa';
+import { ModalProps, Employee } from '../../model';
+import createUserModalStyle from './createEngagementModal.module.scss';
+import { EngagementType } from '../../model/engagementType';
+import { createEngagement } from '../../store/engagementSlice';
+import 'flatpickr/dist/themes/airbnb.css';
+import { selectEmployees } from '../../store/employeeSlice';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface CreateUserConfiguration {}
+
+const CreateEngagementModal = ({ onClose }: ModalProps) => {
+  const dispatch = useDispatch();
+  const { getTokenSilently } = useAuth0();
+  const employeeList = useSelector(selectEmployees);
+  const [employeeId, setEmployeeId] = useState<number>(0);
+  const [startTime, setStartTime] = useState<Date>(new Date());
+  const [endTime, setEndTime] = useState<Date>(new Date());
+  const [engagementType, setEngagementType] = useState<EngagementType>(
+    EngagementType.Shift
+  );
+
+  const handleEmployeeChange = (
+    event: React.ChangeEvent<{ name?: string | undefined; value: unknown }>
+  ) => {
+    setEmployeeId(event.target.value as number);
+  };
+
+  const handleEngagementTypeChange = (
+    event: React.ChangeEvent<{ name?: string | undefined; value: unknown }>
```

Comment:
`name?: string | undefined` - seems like either `?` or `| undefined` is redundant. Also name is never used so might not be required at all.

Is value the type of `event.target.value`? If so might as well give it expected type.

```suggestion
    event: React.ChangeEvent<{ value: EngagementType }>
```